ClassFieldTheory/Cohomology/FiniteCyclic/UpDown.lean

```diff
@@ -419,21 +420,98 @@ end Rep
 include instCyclic in
 def periodicTateCohomology (n : ℤ) :
     tateCohomology (R := R) (G := G) n ≅ tateCohomology (n + 2) :=
-  sorry
+    sorry
+
+
+def periodicTateCohomology' (n : ℤ) (m : ℕ) :
+    tateCohomology (R := R) (G := G) n ≅ tateCohomology (n + 2 * m) := by
+  induction m with
+  | zero => simp; rfl
+  | succ k ih =>
+    apply Iso.trans ih
+    rw [show n + 2 * ↑(k + 1) = n + 2 * k + 2 by omega]
```

Comment:
You can't make data with `rw` either. In general you shouldn't be in tactic mode at all if you're making data, unless you know what you're doing (by which I mean "unless you completely avoid tactics such as `rw` and `simp`).